packages/compiler/core/cli/cli.ts

```diff
@@ -32,6 +32,12 @@ import { CompileCliArgs, getCompilerOptions } from "./args.js";
 async function main() {
   console.log(`TypeSpec compiler v${typespecVersion}\n`);
 
+  const nodeMajorVersion = Number(process.versions?.node?.split(".")?.[0]);
```

Comment:
actually wondering if that check is maybe too late? if it crash because of invalid syntax or import not found.

Reply:
Hmm, maybe. :(

Reply:
Maybe in cmd/tsp.js, would even that fail if node is say so old that you can't `await`?

Reply:
Testing.

Reply:
probably good to have it there, just thinking that if we lets say raise this to 18, could imagine case were you update locally buit not globally and it complains still but I guess that's ok no?

Reply:
(or the other way around where it doesn't complain but crash later)

Reply:
So, this works only node 14. node 12 fails right away with syntax error on top level await in cmd/tsp.js. node 10 and earlier right away on import. It would take significant trickery to get this to work for very old versions. I'm wondering how much is worth it.

Reply:
Pushed a fix with trickery, still not sure if we should bother, though.

Reply:
hhm yeah it is not very pretty, on one side its not something we should have to touch nearly ever(unless bumping version of node) on the other there is already a warning when installing on older version from npm. (+ node 14 and below are not even supported by node anymore)

Reply:
+ I think with this we still have the problem where you might have not updated your global install and so it won't complain and then still crash on the local install